hathor/builder/builder.py

```diff
@@ -180,6 +179,12 @@ def build(self) -> BuildArtifacts:
             **kwargs
         )
 
+        stratum_factory: Optional[StratumFactory] = None
+        if self._enable_stratum_server:
+            stratum_factory = StratumFactory(manager=manager)
+            manager.stratum_factory = stratum_factory
+            manager.metrics.stratum_factory = stratum_factory
+
```

Comment:
I agree, so I moved the code to a function.

I don't think it's worth refactoring HathorManager because it just holds a reference to the stratum factory. It's better to refactor the components that depend on the stratum factory to access it directly. But I don't think it's worth doing it now.